components/style/properties/longhand/background.mako.rs

```diff
@@ -89,108 +89,51 @@ ${helpers.predefined_type("background-color", "CSSColor",
     }
 </%helpers:vector_longhand>
 
-<%helpers:vector_longhand name="background-position-x" animation_type="normal"
+<%helpers:predefined_type name="background-position-x", type="position::HorizontalPosition"
+                          initial_value="
+                              use values::generics::position::HorizontalPosition;
```

Comment:
Instead of this (which feels like abusing these fields), let's add some helper methods to `HorizontalPosition` and `VerticalPosition`, and use them here.